### DISCUSSION TITLE
Foomuuri hangs for one minute on inital start / Devuan

### DISCUSSION THREAD

**Discussion by hpeters** [FoobarOy/foomuuri#32] [General]
I have been trying to run Foomuuri on Devuan (Debian without systemd). Everything has been working fine except when after a reboot the first time you run foomuuri it hangs for one minute.
If i run foomuuri start-or-good(or just start) from the command line it hangs for one minute before completing. If i hit control-c after entering the above command it stops and i notice that the firewall rules have been loaded.
I can then run foomuuri start-or-good after the first time without any problems.
Any idea what could cause that?

**Comment by kimheino** [comment]
Let's try three things:

1) Try removing try-import-except from lines 34-38. Change it to:

```
# SystemD notify support is optional
HAVE_NOTIFY = False
```

I guess that doesn't help, but let's try that first. If it does, then that's the solution.

2) Try running these two commands after boot:

```
foomuuri check   # not start! This will generate the ruleset
nft --file /var/lib/foomuuri/next.fw  # Load ruleset manually
```

That should be fast?

3) When you press Ctrl-C you should see Traceback error.  It will give a hint where it hangs. Copy-paste it to me.

**Comment by hpeters** [comment]
1. Didn't help.
2. foomuuri check # Fast
    nft --file /var/lib/foomuuri/next.fw  # Load ruleset manually  # Hangs.
3. No error. Just prints out start-or-good success.

So it appears to be something with nft.

I tried going back to your basic host bidirectional example.
But it still hangs.

zone {
  localhost
  public eth0
}

public-localhost {
  dhcp-client
  dhcpv6-client
  ping saddr_rate "5/second burst 20"
  ssh saddr_rate "5/minute burst 5"
  drop log
}

localhost-public {
  dhcp-server
  dhcpv6-server
  domain
  http
  https
  imap
  ntp
  ping
  smtp
  ssh
  reject log
}

if I change localhost-public to:

localhost-public {
  accept
}

It works.

**Comment by kimheino** [comment]
> 2. foomuuri check # Fast
>    nft --file /var/lib/foomuuri/next.fw  # Load ruleset manually  # Hangs.

I see. This means that the problem is in netfilter (nft-binary or kernel), not in Foomuuri. You should probably ask netfilter or Devuan people instead.

You might get some clue with `nft --debug all --file /var/lib/foomuuri/next.fw` command, or `dmesg`. For example, does nft trigger starting IPv6 stack?

> I tried going back to your basic host bidirectional example. But it still hangs.
...
> if I change localhost-public to:
> localhost-public { accept }
> It works.

There might be a workaround to this hang if you find out which line creates this problem. Try adding them manually one by one?

**Comment by hpeters** [comment]
It seems there are a lot of different options that can cause the hang.
I did come up with a work around by loading in the bare minimum generated from the config:

 zone {
  localhost
  public eth0
}

public-localhost {
  dhcp-client
  dhcpv6-client
  ssh saddr_rate "5/minute burst 5"
  drop log
}

localhost-public {
  accept
}

If I do that first i can then load in the real firewall without it hanging.

**Comment by hpeters** [comment]
Moved testing to a real server rather than a local virtual machine and it doesn't have the problem
